packages/nuxt-locokit/src/stores/users.ts

```diff
@@ -1,175 +1,57 @@
     sort = {
       createdAt: -1,
     },
   ) {
     loading.value = true
     error.value = null
-    try {
-      users.value = await sdkClient.service('user').find({
-        query: {
-          params,
-          // $sort: sort,
-        },
-      })
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.error(err)
-      error.value = err as Error
-    }
-    loading.value = false
-  }
-
-  async function searchUsers({
-    query,
-    filters,
-    pageIndex = 0,
-    limit = ITEMS_PER_PAGE,
-    sort = {
-      createdAt: -1,
-    },
-  }: {
-    query?: string | null
-    filters?: Filter[] | null
-    pageIndex?: number
-    limit?: number
-    sort?: Record<string, number> | null
-  }) {
-    loading.value = true
-    error.value = null
-    try {
-      let result = null
-      if (filters && query) {
-        result = await sdkClient.service('user').find({
-          query: {
-            username: {
-              $ilike: `%${query}%`,
-            },
-            ...getCurrentFilters(filters),
-            $limit: limit,
-            $skip: pageIndex * limit,
-            // $sort: sort,
-          },
-        })
-      } else if (query) {
-        result = await sdkClient.service('user').find({
-          query: {
-            username: {
-              $ilike: `%${query}%`,
-            },
-            $limit: limit,
-            $skip: pageIndex * limit,
-            // $sort: sort,
-          },
-        })
-      } else if (filters) {
-        result = await sdkClient.service('user').find({
-          query: {
-            ...getCurrentFilters(filters),
-            $limit: limit,
-            $skip: pageIndex * limit,
-            // $sort: sort,
-          },
-        })
-      }
-      loading.value = false
-      return result
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.error(err)
-      error.value = err as Error
+    const res = await findUsers(params, sort)
+    if (res instanceof Error) {
+      error.value = res
+    } else {
+      users.value = res
     }
     loading.value = false
   }
 
   async function blockAccountUser(id: string, blocked: boolean) {
     loading.value = true
     error.value = null
-    try {
-      const res = await sdkClient.service('user').patch(id, {
-        blocked: !blocked,
-      })
-      loading.value = false
-      return res
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.error(err)
-      error.value = err as Error
+    const res = await patchUser(id, {
+      blocked: !blocked,
+    })
+    if (res instanceof Error) {
+      error.value = res
+    } else {
+      users.value = res
     }
     loading.value = false
   }
 
-  function squashUsers(data: {
+  async function squashUsers(data: {
     id: string
     username: string
     lastName: string | null
     firstName: string | null
   }) {
-    if (users.value.total > 0) {
+    if (users.value && users.value.total > 0) {
```

Comment:
I think the notation `users.value?.total` is correct here.

Reply:
I am agree with you but not ts.
`TS18048: 'users.value.total' is possibly 'undefined'.`